src/test/lib/decorators_test.ts

```diff
@@ -389,6 +391,66 @@ suite('decorators', () => {
     });
   });
 
+  suite('@queryAssignedNodes', () => {
+    @customElement('assigned-nodes-el' as keyof HTMLElementTagNameMap)
+    class D extends LitElement {
+      @queryAssignedNodes() defaultAssigned!: Node[];
+
+      @queryAssignedNodes('footer', true) footerAssigned!: Node[];
+
+      render() {
+        return html`
+          <slot></slot>
+          <slot name="footer"></slot>
+        `;
+      }
+    }
+
+    @customElement(generateElementName() as keyof HTMLElementTagNameMap)
+    class C extends LitElement {
+      @query('div') div!: HTMLDivElement;
+      @query('assigned-nodes-el') assignedNodesEl!: D;
+
+      render() {
+        return html`
+          <assigned-nodes-el><div>A</div><slot slot="footer"></slot></assigned-nodes-el>
+        `;
+      }
+    }
+
+    test('returns assignedNodes slot', async () => {
+      const c = new C();
+      container.appendChild(c);
+      await c.updateComplete;
+      await c.assignedNodesEl.updateComplete;
+      assert.deepEqual(c.assignedNodesEl.defaultAssigned, [c.div]);
+      const child = document.createElement('div');
+      c.assignedNodesEl.appendChild(child);
```

Comment:
Can you add some comments? I'm unsure what's being tested here. If you're assigning elements directly into the `assignedNodesEl`, then what is the slot for in C?

Reply:
Done. One set of elements is used to test both the flatten and non-flattened cases. Added comments to explain.